lua/neogit/status.lua

```diff
@@ -64,390 +69,191 @@ local function get_section_item_for_line(linenr)
     return nil, nil
   end
 
-  return section, status[section.name][item_idx]
+  return section, section.files[item_idx]
 end
 
 local function get_current_section_item()
   return get_section_item_for_line(vim.fn.line("."))
 end
 
-local function toggle_sign_at_line(line)
-  local sign_info = status_buffer:get_sign_at_line(line, "fold_markers")
-  local sign = sign_info.signs[1]
-
-  if sign ~= nil then
-    local parts = vim.split(sign.name, ":")
-    local new_name = (parts[1] == "NeogitOpen" and "NeogitClosed" or "NeogitOpen") .. ":" .. parts[2]
-    status_buffer:place_sign(line, new_name, 'fold_markers')
-  end
-end
-
-local function toggle()
-  vim.cmd("silent! normal za")
-
-  if not config.values.disable_signs then
-    local section, item = get_current_section_item()
-
-    if section == nil then
-      return
-    end
-
-    local line = item ~= nil and item.first or section.first
-
-    local on_hunk = item ~= nil and line_is_hunk(vim.fn.getline('.'))
-
-
-    if on_hunk then
-      local ignored_sections = { "untracked_files", "stashes", "unpulled", "unmerged" }
-
-      for _, val in pairs(ignored_sections) do
-        if val == section.name then
-          return
-        end
+local function new_output()
```

Comment:
Same thing as with `collect`. Could we also do some metatable magic here like with collect? Also the name `output` seems very specific even though the implementation could be reused easily. Isn't this more something like a string buffer of builder?